packages/core/admin/admin/src/content-manager/pages/EditView/index.js

```diff
@@ -47,44 +47,28 @@ const EditView = ({
   const { trackUsage } = useTracking();
   const { formatMessage } = useIntl();
   const { createActionAllowedFields, readActionAllowedFields, updateActionAllowedFields } =
-    useMemo(() => {
-      return getFieldsActionMatchingPermissions(userPermissions, slug);
-    }, [userPermissions, slug]);
+    getFieldsActionMatchingPermissions(userPermissions, slug);
 
-  const configurationPermissions = useMemo(() => {
-    return isSingleType
-      ? cmPermissions.singleTypesConfigurations
-      : cmPermissions.collectionTypesConfigurations;
-  }, [isSingleType]);
+  const configurationPermissions = isSingleType
+    ? cmPermissions.singleTypesConfigurations
+    : cmPermissions.collectionTypesConfigurations;
 
   // // FIXME when changing the routing
   const configurationsURL = `/content-manager/${
     isSingleType ? 'singleType' : 'collectionType'
   }/${slug}/configurations/edit`;
   const currentContentTypeLayoutData = get(layout, ['contentType'], {});
 
-  const DataManagementWrapper = useMemo(
-    () => (isSingleType ? SingleTypeFormWrapper : CollectionTypeFormWrapper),
-    [isSingleType]
-  );
+  const DataManagementWrapper = isSingleType ? SingleTypeFormWrapper : CollectionTypeFormWrapper;
 
   // Check if a block is a dynamic zone
-  const isDynamicZone = useCallback((block) => {
+  const isDynamicZone = (block) => {
     return block.every((subBlock) => {
       return subBlock.every((obj) => obj.fieldSchema.type === 'dynamiczone');
     });
-  }, []);
+  };
 
-  const formattedContentTypeLayout = useMemo(() => {
-    if (!currentContentTypeLayoutData.layouts) {
-      return [];
-    }
-
-    return createAttributesLayout(
-      currentContentTypeLayoutData.layouts.edit,
-      currentContentTypeLayoutData.attributes
-    );
-  }, [currentContentTypeLayoutData]);
+  const formattedContentTypeLayout = createAttributesLayout(currentContentTypeLayoutData);
```

Comment:
@joshuaellis I'd like you opinion here: this is a case where I'd assume `useMemo` would be useful, because `createAttributesLayout` does quite a bit of looping.

Reply:
It does a fair bit of looping, it's hard to know without measuring.

*if* we were to do a memo, I would do this because I think `layout` might update per render –

```ts
const EditView = ({layout}) => {
//...
	const formattedContentTypeLayout = useMemo(() => {
		  const currentContentTypeLayoutData = get(layout, ['contentType'], {});
			return createAttributesLayout(currentContentTypeLayoutData)
	}, [layout])
//...
}
```

But looking at the code, we pass `layout` from `EditViewDataManager` using `useSelector` and pass it 2/3 levels down, why? why don't we instead select this `layout` state in the `EditView` component and return that as well as doing this calculation and let `redux` handle the memoization?

Reply:
So really, when we tackle these issues we could really be addressing other issues in the codebase e.g prop drilling across pages where it's not totally necessary when we can isolate the pages :)

Reply:
Thanks for your feedback!

@gu-stav https://github.com/strapi/strapi/pull/14881/commits/53166af4fe19a4463f9476563872b1a641f55858
@joshuaellis https://github.com/strapi/strapi/pull/14881/commits/90975c5d93226319887e048fa1dbfa1eb471cf17